contracts/upgradeable_contracts/ChaiConnector.sol

```diff
@@ -0,0 +1,156 @@
+pragma solidity 0.4.24;
+
+import "../interfaces/IChai.sol";
+import "./Ownable.sol";
+import "./ERC20Bridge.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+* @title ChaiConnector
+* @dev This logic allows to use Chai token (https://github.com/dapphub/chai)
+*/
+contract ChaiConnector is Ownable, ERC20Bridge {
+    using SafeMath for uint256;
+
+    bytes32 internal constant CHAI_TOKEN = 0xe529dd1fa310362a861f9a51ed0d07b46ef28d89054300cd2734814ddfcfd449; // keccak256(abi.encodePacked("chaiToken"))
+    bytes32 internal constant INVESTED_AMOUNT = 0xb6afb3323c9d7dc0e9dab5d34c3a1d1ae7739d2224c048d4ee7675d3c759dd1b; // keccak256(abi.encodePacked("investedAmount"))
+
+    uint256 internal constant ONE = 10**27;
+
+    /**
+    * @dev Fixed point multiplication
+    * @return Truncated value of x * y
+    */
+    function rmul(uint256 x, uint256 y) internal pure returns (uint256) {
+        return x.mul(y) / ONE;
+    }
+
+    /**
+    * @dev Fixed point division
+    * @return Ceiled value of x / y
+    */
+    function rdivup(uint256 x, uint256 y) internal pure returns (uint256) {
+        return x.mul(ONE).add(y.sub(1)) / y;
+    }
+
+    /**
+    * @dev Initializes chai token
+    * @param _chai Chai token contract address
+    */
+    function initializeChaiToken(address _chai) external onlyOwner {
+        require(address(chaiToken()) == address(0));
+        require(address(IChai(_chai).daiToken()) == address(erc20token()));
+        addressStorage[CHAI_TOKEN] = _chai;
+    }
+
+    /**
+    * @dev Withdraws all invested tokens, pays remaining interest, removes chai token from contrac storage
+    * @param recipient Account address to receive remaining interest
+    */
+    function removeChaiToken(address recipient) external onlyOwner {
+        _convertChaiToDai(investedAmountInDai());
+        chaiToken().transfer(recipient, chaiBalance());
+        delete addressStorage[CHAI_TOKEN];
+    }
+
+    /**
+    * @dev Pays all available interest, in Chai tokens
+    * @param recipient Account address to receive available interest
+    */
+    function payInterest(address recipient) external onlyOwner {
```

Comment:
Having in mind the scheme required by the PoS introduction in the future (https://www.staketoken.net/token-model/xdai-stable-token) consider to do the following:
  * change the existing method `payInterest` as so to be available for every one. This method must pay the interest on a configured account (make sure that it is not zero). The method must allow its invocation with some minimal period (e.g. not often than once per week). If the method is called by the **upgradability owner** (as `claimTokens` in `BasicBridge.sol`) the rule to allow paying the interest with some minimal period is not applied. As soon as the interest payed, the configured account (if it is a contract) must be notified by calling a method (it seems logical to have `onTokenTransfer` there).
  * implement a method available for the owner to set up the account to receive the interest.
  * implement a method to configure min period between two subsequent attempts to call the method to pay the interest.

Reply:
Should we pay the interest in Dai directly, so than the bridge will have to withdraw Dai first?

Reply:
Thanks for the question. My view is that the exchange from Chai to Dai will be responsibility of the receiver (either EoA or a contract). Do you see any objections?

Reply:
If the users are not going to operate with Chai token directly, then I think it is better to keep all the functionality related to Chai in one place. 
This will also simplify switching to other investing approach in future updates, if needed.

Reply:
Good reasoning. Let's implement the functionality in the way you suggested. Thanks!